### ISSUE TITLE
Create database on initial migration?

### ISSUE THREAD

**Issue by jasonwilson** [mikro-orm/mikro-orm#1757]
**Is your feature request related to a problem? Please describe.**
In trying to setup a new database from an initial migration I realized there's no way to create a new database from scratch. Is that accurate?

**Describe the solution you'd like**
Something akin to rake db:init or rake db:create where mikro-orm will take the runtime config and create a new database if it doesn't exist

**Describe alternatives you've considered**
Tried creating a migration to do this for me but mikro-orm migrator fails to initialize because the database doesn't exist. 

```
  async up(): Promise<void> {
    this.addSql(`CREATE DATABASE ${config.dbName};`)
  }
```

**Comment by B4nan** [comment]
There is `SchemaGenerator.ensureDatabase()` you could call programatically. It is automatically called when you use the schema generator from CLI too (e.g. creating schema always first ensures the db exists). Not sure if it's good idea to add this to migrations, the feature on its own feels quite distinct. But we can definitely expose that method via CLI (as one of the schema generator commands, so again, not connected to migrations).

> Something akin to rake db:init or rake db:create where mikro-orm will take the runtime config and create a new database if it doesn't exist

Never used rails, so please actually describe what you want to propose, as this has no meaning to me :P But both of that sounds like they work more like schema generator rather than migrations (and we already have that).

**Comment by jasonwilson** [comment]
Good point on conflating db creation + migration that could be a little confusing. Saying that, tying that action specifically to  schema creation also feels disconnected... 

In rails land `rake db:create` will initialize a new database with the supplied  database configuration. If mikro-orm exposed a command like `mikro-orm database:create` to initialize a new database with the supplied configuration that would be helpful. This would allow for users to spinup a db from scratch without leaving the ORM. Thoughts?

For now I'll experiment with creating a helper using `SchemaGenerator.ensureDatabase()`

**Comment by B4nan** [comment]
> This would allow for users to spinup a db from scratch without leaving the ORM. 

This sounds more like you want the schema to be generated too, otherwise you will have just clear db. If so, then again, schema generator is the way to go and it already supports creating db automatically (`mikro-orm schema:create -r`).

**Comment by jasonwilson** [comment]
Not quite- in this case we'll have the migrations so the flow would be:
1. Create database
2. Run migrations

**Comment by B4nan** [comment]
I think we could just create the db right ahead, just like we do in the schema generator.

Looking at the code and `migration` commands (at least `up/down`) will first ensure the migrations table exists, as they requite to read the data from migrations table. But it only checks if the table exists (and creates it if not), so we should just add the `ensureDatabase` call before that. 

But I still see a value in having the separate command to create the db.

PR welcome, not a big priority but if nobody will pick this up, will do so myself before v5 is out.
